airflow/models/dag.py

```diff
@@ -1524,12 +1524,12 @@ def bulk_sync_to_db(cls, dags: Collection["DAG"], sync_time=None, session=None):
             orm_dag.description = dag.description
             orm_dag.schedule_interval = dag.schedule_interval
             for orm_tag in list(orm_dag.tags):
-                if orm_tag.name not in orm_dag.tags:
+                if orm_tag.name not in set(dag.tags):
```

Comment:
Change Iterable to set as discuss in https://github.com/apache/airflow/pull/8233#issuecomment-612643623